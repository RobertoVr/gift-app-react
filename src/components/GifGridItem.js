import React from 'react'

export const GifGridItem = ({ id, title, url }) => {
    return (
        <div>
            <p>{title}</p>
            <img src={url} alt="gif" />
        </div>
    )
}
