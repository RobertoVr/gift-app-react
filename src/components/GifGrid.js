import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    useEffect(() => {
        getGifs();
    }, []);

    const getGifs = async () => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Goku&limit=10&api_key=PVTSe8E8rAfl36W4IWV3d2GoSMEkLoVh';
        const respuesta = await fetch(url);
        const { data } = await respuesta.json();
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        setImages(gifs);
    }

    return (
        <div>
            {
                images.map((img) => (
                    <GifGridItem
                        key={img.id}
                        {...img}
                    />
                ))
            }
        </div>
    )
}
