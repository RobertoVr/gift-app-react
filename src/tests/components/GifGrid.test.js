import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas sobre el componente <GifGrid />', () => {
    const category = 'goku';

    test('Debe de hacer correctamente el render', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar items cuando cargan imagenes', () => {
        const imgs = [
            {
                id: 'BASHJG',
                title: 'Imagen prueba',
                url: 'https://localhost/123.png'
            },
            {
                id: '12BASHJG',
                title: 'Imagen prueba',
                url: 'https://localhost/123.png'
            }
        ];

        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(imgs.length);
    })


})
