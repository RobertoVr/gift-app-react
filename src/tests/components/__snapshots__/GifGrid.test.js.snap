// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GifGrid /> Debe de hacer correctamente el render 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    goku
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas sobre el componente <GifGrid /> debe de mostrar items cuando cargan imagenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="BASHJG"
      key="BASHJG"
      title="Imagen prueba"
      url="https://localhost/123.png"
    />
    <GifGridItem
      id="12BASHJG"
      key="12BASHJG"
      title="Imagen prueba"
      url="https://localhost/123.png"
    />
  </div>
</Fragment>
`;
